tsdb/querier.go

```diff
@@ -407,6 +407,20 @@ func labelValuesWithMatchers(r IndexReader, name string, matchers ...*labels.Mat
 	return values, nil
 }
 
+func labelNamesWithMatchers(r IndexReader, matchers ...*labels.Matcher) ([]string, error) {
+	p, err := PostingsForMatchers(r, matchers...)
+	if err != nil {
+		return nil, err
+	}
+
+	var postings []uint64
```

Comment:
This could potentially be the list of all series in the block, like if the only matcher we pass in matches all the series. But I think we'll save important roundtrips later in `r.LabelNamesFor` by grabbing them all here.